src/events/interactionCreate.ts

```diff
@@ -301,13 +303,18 @@ async function handleButtonInteraction(
 
 	logger.debug(`Calling button handler '${button.customId}'`);
 
-	const buttonContext: ButtonContext = {
+	/* eslint-disable @typescript-eslint/consistent-type-assertions */
+	const buttonContext = {
```

Comment:
This declaration can be rewritten to exclude the assertion and ESLint disablements:
```ts
const buttonContext: ButtonContext = {
	...context,
	interaction,
	component: interaction.component,
	message: interaction.message,
	channelId: interaction.channelId,
};
```

Reply:
done